src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,38 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
```

Comment:
Code quality issue: It's recommended to use try-with-resources for Session to make sure it's always closed. Even though you have a finally block that closes the session, using try-with-resources can make the code cleaner and more compact.